apollo-backend/src/main/java/io/logz/apollo/controllers/ScalingController.java

```diff
@@ -82,6 +90,24 @@ public void updateScalingFactor(int groupId, int scalingFactor, Req req) {
         group.setScalingStatus(Group.ScalingStatus.PENDING);
         groupDao.updateGroup(group);
 
+        String userEmail = req.token(Tokens._USER).toString();
+        Service service = serviceDao.getService(group.getServiceId());
+        Environment environment = environmentDao.getEnvironment(group.getEnvironmentId());
+
+        MDCLogging.withMDCFields(ImmutableMap.of("userEmail", userEmail,
+                                                "groupName", group.getName(),
+                                                "serviceName", service.getName(),
+                                                "envName", environment.getName(),
+                                                "newScalingFactor", String.valueOf(scalingFactor)),
+            () -> logger.info("Scaling factor changed from {} to {} for group {} by user email {}, env ID: {}, service ID: {}, group ID: {}",
```

Comment:
You can remove the groupId we actually use the groupName

Reply:
I think that environmentId, serviceId and groupId are better

Reply:
For me, it's hard to rely on ID alone since it forces me to examine the DB